tests/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.24)
+
+set(TEST_NAME JetModuleTests)
+
+include(FetchContent)
+
+set(gtest_REQUIREDVERSION "1.12.0")
+
+find_package(GTest ${gtest_REQUIREDVERSION} )
+
+if(GTest_FOUND)
```

Comment:
It is better to fetch every time gtest and use it under the _deps folder. So just fetch it. Because then you ensure that your test is every time working / compiling with the right gtest version.